contracts/ECDSA.sol

```diff
@@ -0,0 +1,146 @@
+pragma solidity ^0.6.0;
+
+
+/**
+ * @dev Elliptic Curve Digital Signature Algorithm (ECDSA) operations.
+ *
+ * These functions can be used to verify that a message was signed by the holder
+ * of ECDSA keys corresponding to a certain account.
+ *
+ * Accounts may register
+ */
+contract NearECDSA {
+
+    constructor () public {}
+
+    mapping (address => address) public accounts;
+
+    function registerSigner(address signer) external {
+        accounts[signer] = msg.sender;
+    }
+
+    function unregisterSigner(address signer) external {
+        if (accounts[signer] = msg.sender) accounts[signer] = address(0);
+    }
+
+    function recover(bytes32 _hash, bytes memory signature) external view returns (address) {
```

Comment:
🤔  wrong params? in fact this whole function is quite confusing.

Reply:
oh I missed a step

Reply:
now we actually recover in the function